packages/vite-plugin-cloudflare/src/runner-worker/index.ts

```diff
@@ -169,65 +218,67 @@ export function createWorkerEntrypointWrapper(
 				const request = arg as Request;
 				const url = new URL(request.url);
 
-				let webSocket: WebSocket;
+				// Initialize the module runner
 				if (url.pathname === INIT_PATH) {
-					try {
-						const viteDevMetadata = getViteDevMetadata(request);
-						entryPath = viteDevMetadata.entryPath;
-						const { 0: client, 1: server } = new WebSocketPair();
-						webSocket = client;
-						await createModuleRunner(this.env, server);
-					} catch (e) {
-						return new Response(
-							e instanceof Error ? e.message : JSON.stringify(e),
-							{ status: 500 }
+					const workerEntryPathHeader = request.headers.get(
+						WORKER_ENTRY_PATH_HEADER
+					);
+
+					if (!workerEntryPathHeader) {
+						throw new Error(
+							`Unexpected error: "${WORKER_ENTRY_PATH_HEADER}" header not set.`
 						);
 					}
 
-					return new Response(null, {
-						status: 101,
-						webSocket,
-					});
+					// Set the Worker entry path
+					workerEntryPath = workerEntryPathHeader;
+					const stub = this.env.__VITE_RUNNER_OBJECT__.get("singleton");
+
+					// Forward the request to the Durable Object to initialize the module runner and return the WebSocket
+					return stub.fetch(request);
 				}
 			}
 
-			const entrypointValue = await getWorkerEntryExport(entryPath, entrypoint);
+			const exportValue = await getWorkerEntryExport(
```

Comment:
Could we move these functions inside the DO and access them via RPC like:

```suggestion
			const stub = this.env.__VITE_RUNNER_OBJECT__.get("singleton");
			const exportValue = await stub.getWorkerEntryExport(
```

Reply:
and so on to allow us to move the moduleRunner object itself into the DO?

Reply:
Even if this was possible, `moduleRunner.import()` isn't the only that the module runner is accessed. Modules are also fetched via the module runner when running the user code.